cdk_framework/EKS/lib/resource_constructs/general_constructs/test-case-resource-deployment-construct.ts

```diff
@@ -0,0 +1,79 @@
+import { Construct } from 'constructs';
+import { Cluster, FargateCluster } from 'aws-cdk-lib/aws-eks';
+import { NamespaceConstruct } from '../specific_constructs/namespace-construct';
+import { GeneralSampleAppDeploymentConstruct, GeneralSampleAppDeploymentConstructProps } from './general-sample-app-deployment-construct';
+import { GeneralCollectorDeploymentConstruct, GeneralCollectorDeploymentConstructProps } from './general-collector-deployment-construct';
+
+export class TestCaseResourceDeploymentConstruct extends Construct {
+    name: string
+    namespace: Construct
+
+    constructor(scope: Construct, id: string, props: TestCaseResourceDeploymentConstructProps){
+        super(scope, id);
+
+        // namespace deployment
+        const namespaceName = `collector-namespace`
+        const collectorNamespaceConstruct = new NamespaceConstruct(this, 'collector-namespace-construct', {
+            cluster: props.cluster,
+            name: namespaceName
+        })
+
+        // used for when sample app is push mode
+        const deployServices = props.sampleAppMode === 'push'
+        const grpcServiceName = 'collector-grpc'
+        const grpcPort = 4317
+        const udpServiceName = 'collector-udp'
+        const udpPort = 55690
+        const tcpServiceName = 'collector-tcp'
+        const httpPort = 4318
```

Comment:
If these are fixed values do they belong in this scope? Could we instead move them down a level, this would prevent us from having to pass them in as props to the to the sample app construct.

Reply:
Since the collector needs these values we can just export them from the construct and reference them that way.

Reply:
Or we could pass in the sample app as a prop to the collector and allow the collector construct to handle those values.

Reply:
I wasn't sure. On the one hand that is a lot cleaner, but if we're looking at the sample app from a more zoomed out perspective, would we perhaps want to allow users to configure the sample app the way that they want?